File: pathpeer-web/src/features/auth/api/authApi.ts
import client from '../../../shared/api/axiosClient';
import type { User } from '../../../shared/types';

interface LoginDto {
    email: string;
    password: string;
}

interface RegisterDto {
    email: string;
    password: string;
    username: string;
}

export const authApi = {
    login: (dto: LoginDto): Promise<User> =>
        client.post('/auth/login', dto).then(r => r.data),

    register: (dto: RegisterDto): Promise<User> =>
        client.post('/auth/register', dto).then(r => r.data),

    logout: (): Promise<void> =>
        client.post('/auth/logout').then(r => r.data),

    getProfile: (): Promise<User> =>
        client.get('/auth/profile').then(r => r.data),
}

File: pathpeer-mobile/lib/features/auth/data/auth_api.dart
import 'package:dio/dio.dart';
import 'package:pathpeer_mobile/core/network/dio_client.dart';

class AuthApi {
  Future<Map<String, dynamic>> login(String email, String password) async {
    final response = await dio.post(
      '/auth/login',
      data: {'email': email, 'password': password},
      options: Options(
        headers: {'X-Client-Type': 'mobile'},
      ),
    );
    return response.data;
  }

  Future<Map<String, dynamic>> register(
    String email,
    String password,
    String username,
  ) async {
    final response = await dio.post(
      '/auth/register',
      data: {
        'email': email,
        'password': password,
        'username': username,
      },
      options: Options(
        headers: {'X-Client-Type': 'mobile'},
      ),
    );
    return response.data;
  }

  Future<void> logout() async {
    await dio.post('/auth/logout');
  }

  Future<Map<String, dynamic>> getProfile() async {
    final response = await dio.get('/auth/profile');
    return response.data;
  }
}